src/gf.c

```diff
@@ -1084,20 +1090,20 @@ static int check_ambiguous_visitor(jl_typemap_entry_t *oldentry, struct typemap_
     jl_method_t *m = closure->newentry->func.method;
     jl_tupletype_t *sig = oldentry->sig;
     jl_value_t *isect = closure->match.ti;
-    if (sigs_eq(isect, (jl_value_t*)(closure->after ? sig : type), 1)) {
+    if (jl_types_equal(isect, (jl_value_t*)(closure->after ? sig : type))) {
         // we're ok if the new definition is actually the one we just
```

Comment:
I feel like this comment is wrong. It seems to be based on the premise that if `New ⊓ Old == New` then `New <: Old`, but that's obviously wrong (although it was sort-of useful as a heuristic in the old type system with eager ambiguity printing). I believe this conditional should go away now, or be converted into an actual subtyping test (if that's faster / more accurate than just relying on `jl_type_morespecific` returning correct answers)